packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/lib/LinkQueueLogger.ts

```diff
@@ -0,0 +1,164 @@
+import type { ILinkQueue, ILink } from '@comunica/bus-rdf-resolve-hypermedia-links-queue';
+import { LinkQueueWrapper } from '@comunica/bus-rdf-resolve-hypermedia-links-queue';
+import type { Logger } from '@comunica/types';
+
+/**
+ * A link queue that log information about event happening in the queue
+ */
+export class LinkQueueLogger extends LinkQueueWrapper {
+  public readonly query: string;
+  private readonly logger: Logger;
+  private readonly linkProductionRatio: ILinkProductionActorRatio = {
+    pushEvents: {},
+    popEvents: {},
+  };
+
+  public static readonly LINK_QUEUE_EVENT_NAME = 'Link queue changed';
+
+  /**
+   *
+   * @param {ILinkQueue} linkQueue - The link queue
+   * @param {Algebra.Operation} query - The current query
+   * @param {Logger} logger - The logger where the events are output
+   */
+  public constructor(linkQueue: ILinkQueue, query: string, logger: Logger) {
+    super(linkQueue);
+    this.logger = logger;
+    this.query = query;
+  }
+
+  /**
+   * Helper function to get the reachability criteria of a link
+   * @param {ILink} link - Current link
+   * @returns {IProducedByActor | null } The reachability criteria with extra information about it if available
+   */
+  private static getActorProductorInformation(link: ILink): IProducedByActor | null {
+    const metadata = link.metadata;
+    if (metadata !== undefined && metadata[PRODUCED_BY_ACTOR] !== undefined) {
+      const { name, ...rest } = metadata[PRODUCED_BY_ACTOR];
+
+      if (name === undefined) {
+        return null;
+      }
+      return {
+        name,
+        metadata: Object.keys(rest).length === 0 ? undefined : rest,
+      };
+    }
+    return null;
+  }
+
+  /**
+   * Update the link production ratio
+   * @param {IUrlStatistic} link - current link
+   * @param {keyof ILinkProductionActorRatio} event - link queue event
+   */
+  private updateLinkProductionRatio(link: IUrlStatistic, event: keyof ILinkProductionActorRatio): void {
+    if (link.producedByActor &&
+      this.linkProductionRatio[event][link.producedByActor?.name]) {
+      this.linkProductionRatio[event][link.producedByActor?.name] += 1;
+    } else if (
```

Comment:
This looks wrong. An `if` without actual statement.

Reply:
I don't understand the statement is `this.linkProductionRatio[event][link.producedByActor?.name ?? 'unknown'] = 1;` I think?